src/algebra/big_operators/order.lean

```diff
@@ -304,6 +304,18 @@ lemma sum_card [fintype α] (h : ∀ a, (B.filter $ (∈) a).card = n) :
   ∑ s in B, s.card = fintype.card α * n :=
 by simp_rw [fintype.card, ←sum_card_inter (λ a _, h a), univ_inter]
 
+lemma card_le_card_bUnion {s : finset ι} {f : ι → finset α} (h₀ : ∀ i ∈ s, (f i).nonempty)
+  (h₁ : (s : set ι).pairwise_disjoint f) :
+  s.card ≤ (s.bUnion f).card :=
+by { rw [card_bUnion h₁, card_eq_sum_ones], exact sum_le_sum (λ i hi, (h₀ i hi).card_pos) }
+
+lemma card_le_card_bUnion_add_one {s : finset (finset α)}
+  (hs : (s : set (finset α)).pairwise_disjoint id) :
```

Comment:
Isn't it possible to have this version with `f` rather than `id` like the previous?

Reply:
No. This lemma uses the fact that the fiber of `∅` along `id` has cardinality at most `1`. Imagine `f` sending everything to `∅`.

Reply:
But we have the generalisation where instead of adding 1, we add the cardinality of the fiber?

Reply:
Eh, sure.

Reply:
Feel free to leave this generalisation for a later PR if you prefer by the way, I really only just need the special case you have!

Reply:
Lovely generalizations 🥳